README.md
# ubuntu-18.04-pip3

Vagrant Box with Ubuntu 18.04 and pip3

## Base image

Used base image [elegoev/ubuntu-18.04](https://app.vagrantup.com/elegoev/boxes/ubuntu-18.04)

## Directory Description

| directory | description                                          |
|-----------|------------------------------------------------------|
| inspec    | inspec test profiles with controls                   |
| packer    | packer build, provisioner and post-processor scripts |
| test      | test environment for provision & inspec development  |

## Vagrant

### Vagrant Cloud

- [elegoev/ubuntu-18.04-pip3](https://app.vagrantup.com/elegoev/boxes/ubuntu-18.04-pip3)

### Vagrant Plugins

- [vagrant-disksize](https://github.com/sprotheroe/vagrant-disksize)
- [vagrant-hosts](https://github.com/oscar-stack/vagrant-hosts)
- [vagrant-secret](https://github.com/tcnksm/vagrant-secret)
- [vagrant-vbguest](https://github.com/dotless-de/vagrant-vbguest)
- [vagrant-serverspec](https://github.com/vvchik/vagrant-serverspec)
- [vagrant-vmware-esxi](https://github.com/josenk/vagrant-vmware-esxi)

### Vagrantfile

    Vagrant.configure("2") do |config|

      $basebox_name="ubuntu-18.04-pip3-test"
      $basebox_hostname="ubuntu-1804-pip3-test"
      $src_image_name="elegoev/ubuntu-18.04-pip3"
      $vb_group_name="basebox-pip3-test"

      config.vm.define "#{$basebox_name}" do |machine|
        machine.vm.box = "#{$src_image_name}"
    
        # define guest hostname
        machine.vm.hostname = "#{$basebox_hostname}"

        machine.vm.provider "virtualbox" do |vb|
          vb.name = $basebox_name
          vb.cpus = 1
          vb.customize ["modifyvm", :id, "--memory", "1024" ]
          vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
          vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
          vb.customize [
            "modifyvm", :id, "--uartmode1", "file",
            File.join(Dir.pwd, "ubuntu-bionic-18.04-cloudimg-console.log")
          ]
          vb.customize ["modifyvm", :id, "--groups", "/#{$vb_group_name}" ]
          vb.customize ["modifyvm", :id, "--vram", 256 ]
        end

      end   

    end

## Versioning

Repository follows sematic versioning  [![semantic versioning](https://img.shields.io/badge/semver-2.0.0-green.svg)](http://semver.org)

## Changelog

For all notable changes see [CHANGELOG](https://github.com/elegoev/basebox-ubuntu-18.04-pip3/blob/main/CHANGELOG.md)

## License

Licensed under The MIT License (MIT) - for the full copyright and license information, please view the [LICENSE](https://github.com/elegoev/basebox-ubuntu-18.04-pip3/blob/main/LICENSE) file.

## Issue Reporting

Any and all feedback is welcome.  Please let me know of any issues you may find in the bug tracker on github. You can find it [here.](https://github.com/elegoev/basebox-ubuntu-18.04-pip3/issues)


CHANGELOG.md
# Change Log

## 1.0.x

**Implemented enhancements:**

- Ansible installation

## 2.0.x

**Implemented enhancements:**

- Packer build implemented


packer/post-processors/vagrant-cloud-create-box.ps1
# create vagrant cloud box

# check vagrant cloud token for packer (PACKER_VAGRANTCLOUD_TOKEN)
if (-not (Test-Path env:PACKER_VAGRANTCLOUD_TOKEN)) { 
  Write-Host "Environment variable 'PACKER_VAGRANTCLOUD_TOKEN' not set" -ForegroundColor Red
} else {

  # set vagrant image name
  $IMAGE_NAME = $env:IMAGE_NAME
  $BOX_NAMESPACE = $env:BOX_NAMESPACE
  $BOX_DESCRIPTION = $env:BOX_DESCRIPTION
  $BOX_SHORTDESCRIPTION = $env:BOX_SHORTDESCRIPTION

  # show used token
  # Write-Host "token = $env:PACKER_VAGRANTCLOUD_TOKEN"

  # check if login user
  vagrant cloud auth login --token $env:PACKER_VAGRANTCLOUD_TOKEN
  if($?) {
    
    # check cloud image
    # vagrant cloud search $IMAGE_NAME --short | Select-String -Pattern 'elegoev/ubuntu-18.04'
    Write-Host "image name = $IMAGE_NAME"
    vagrant cloud box create --no-private "$BOX_NAMESPACE/$IMAGE_NAME" `
                             --description "$BOX_DESCRIPTION" `
                             --short-description "$BOX_SHORTDESCRIPTION"

  } else {
    Write-Host "Wrong vagrant cloud token configured" -ForegroundColor Red
  }

}







inspec/baseline-ubuntu-18.04-pip3/controls/check_python.rb
# copyright: 2021, Urs Voegele

title "check python3"

# check python3 command
control "python3-1.0" do                        # A unique ID for this control
  impact 1.0                                   # The criticality, if this control fails.
  title "check if python3 command exists"      # A human-readable title
  desc "check python3 command"
  describe command('python3 --version') do
    its('exit_status') { should eq 0 }
  end
end

# check pip3 command
control "pip3-1.0" do                          # A unique ID for this control
  impact 1.0                                   # The criticality, if this control fails.
  title "check if pip3 command exists"         # A human-readable title
  desc "check pip3 command"
  describe command('pip3 --version') do
    its('exit_status') { should eq 0 }
  end
end


packer/post-processors/create-git-tag.ps1
# create git tag

$src_image_version = $env:IMAGE_VERSION

$vagrant_file_template = '
Vagrant.configure("2") do |config|

  $basebox_name="ubuntu-18.04-pip3-test"
  $basebox_hostname="ubuntu-1804-pip3-test"
  $src_image_name="elegoev/ubuntu-18.04-pip3"
  $vb_group_name="basebox-pip3-test"

  config.vm.define "#{$basebox_name}" do |machine|
    machine.vm.box = "#{$src_image_name}"
    machine.vm.box_version = "$image_version"
    
    # define guest hostname
    machine.vm.hostname = "#{$basebox_hostname}"

    machine.vm.provider "virtualbox" do |vb|
      vb.name = $basebox_name
      vb.cpus = 1
      vb.customize ["modifyvm", :id, "--memory", "1024" ]
      vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
      vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
      vb.customize ["modifyvm", :id, "--uartmode1", "disconnected" ]
      vb.customize ["modifyvm", :id, "--groups", "/#{$vb_group_name}" ]
      vb.customize ["modifyvm", :id, "--vram", 256 ]
    end
  
    # forwarding ports
    machine.vm.network "forwarded_port", id: "rancher-ssl",  auto_correct: true, protocol: "tcp", guest: 8443, host: 8443, host_ip: "127.0.0.1"

    machine.trigger.after :destroy do |trigger|
      trigger.info = "Cleanup environment"
      trigger.run = {path: "./cleanup-vagrant.ps1"}  
    end

  end   

end
'

# create Vagrantfile for download test
$vagrant_file_expanded = $vagrant_file_template.Replace('$image_version', "$src_image_version")
Set-Content -Path './test/download/Vagrantfile' -Value "$vagrant_file_expanded"

# create git tag
git add --all
git commit -m "new basebox version $src_image_version"
git push 
git tag $src_image_version
git push --tags





